pkg/hooks/hooks.go

```diff
@@ -94,29 +96,83 @@ func runDirectory(dirpath string, environment []string, logger logging.Logger) e
 		if f.IsDir() {
 			continue
 		}
-		logger.WithField("path", fullpath).Infof("Executing hook %s", f.Name())
-		cmd := exec.Command(fullpath)
-		hookOut := &bytes.Buffer{}
-		cmd.Stdout = hookOut
-		cmd.Stderr = hookOut
-		cmd.Env = environment
-		err := cmd.Run()
-		if err != nil {
-			logger.WithErrorAndFields(err, logrus.Fields{
-				"path":   fullpath,
-				"output": hookOut.String(),
-			}).Warnf("Could not execute hook %s", f.Name())
-		} else {
-			logger.WithFields(logrus.Fields{
-				"path":   fullpath,
-				"output": hookOut.String(),
-			}).Debugln("Executed hook")
+
+		timeout := 5
+		h := Hook{fullpath, f.Name(), environment, logger, timeout}
+
+		err := h.RunWithTimeout()
+		htErr, ok := err.(HookTimeoutError)
+		if !ok {
```

Comment:
Yes this is a logic error.  Thanks for catching @anthonybishopric 

> what does this do if err == nil? will it warn?

@petertseng - it's actually kind of odd to me that this even works, isn't it?  I was puzzled so I wrote some example code to figure out the behavior, but it turns out that it's right in the [language reference](https://golang.org/ref/spec#Type_assertions):

> A type assertion used in an assignment or initialization of the special form
>     v, ok = x.(T)
>     v, ok := x.(T)
>     var v, ok = x.(T)
> yields an additional untyped boolean value. The value of ok is true if the assertion holds. Otherwise it is false and the value of v is the zero value for type T. No run-time panic occurs in this case.